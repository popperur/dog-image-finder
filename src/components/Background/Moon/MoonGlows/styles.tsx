import styled, { keyframes } from "styled-components";

export const MoonGlowContainer = styled.div``;

const moonGlowFrames = keyframes`
   from {
     transform: scale(1);
   }
   50% {
     transform: scale(1.1);
   }
   to {
     transform: scale(1);
   }
`;

export const MoonGlowHolder = styled.div<{ $styles?: string[] }>`
  position: absolute;
  background-color: #f1f1f1;
  border-radius: 50%;
  opacity: 0.1;
  transition: opacity 0.3s;
  animation: ${moonGlowFrames} infinite;
  ${props => props.$styles && props.$styles.join(";")};
`;
