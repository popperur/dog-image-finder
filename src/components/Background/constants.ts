export const TOOLTIP_PAUSED =
  "Radio signal from the Moon! This incredible discovery could greatly enhance your experience in browsing for dogs!";
export const TOOLTIP_PLAYING = "Auuuuuu!";

// top, left, width, height, animationDelay
export const STARS = [
  [45, 66, 6, 6, 2],
  [70, 48, 5, 5, 4],
  [4, 44, 3, 3, 1],
  [23, 53, 1, 1, 2],
  [26, 93, 3, 3, 1.3],
  [46, 13, 4, 4, 2.3],
  [66, 34, 2, 2, 1.3],
  [36, 3, 3, 3, 3.3],
  [76, 54, 1, 1, 2.3],
  [56, 24, 3, 3, 1.3],
  [76, 25, 3, 3, 0.3],
  [56, 55, 1, 1, 2.31],
  [36, 35, 3, 3, 1.31],
  [86, 45, 2, 2, 2.31],
  [82, 85, 3, 3, 0.31],
  [81, 15, 1, 1, 1.31],
  [83, 35, 3, 3, 2.31],
  [23, 55, 2, 2, 4],
  [53, 50, 1, 1, 1.31],
  [83, 20, 1, 1, 1.51],
];

// top, left, width, height, opacity
export const MOON_CRATERS = [
  [23, 33, 17, 17, 0.5],
  [48, 10, 30, 30, 0.7],
  [90, 14, 20, 20, 0.4],
  [18, 63, 25, 25, 0.6],
  [33, 105, 10, 10, 0.8],
  [82, 55, 14, 14, 0.9],
  [122, 34, 20, 20, 0.6],
  [102, 84, 16, 16, 0.8],
  [63, 113, 20, 20, 0.9],
  [116, 119, 12, 12, 0.6],
];

// top, left, width, height, animationDuration, animationDelay
export const MOON_GLOWS = [
  [20, 30, 260, 260, 3, 0],
  [-20, -10, 340, 340, 3, 0.5],
];

// bottom, left, width, height
export const HILLS = [
  [183, 120, 20, 239],
  [259, 229, 28, 255],
  [211, 334, 28, 246],
];
