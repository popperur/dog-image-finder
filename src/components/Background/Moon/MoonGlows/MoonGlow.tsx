import { MoonGlowHolder } from "components/Background/Moon/MoonGlows/styles.tsx";

interface MoonGlowProps {
  top: number;
  left: number;
  width: number;
  height: number;
  animationDuration: number;
  animationDelay: number;
}

function MoonGlow({
  top,
  left,
  width,
  height,
  animationDuration,
  animationDelay,
}: MoonGlowProps) {
  const stylesArray = [
    `top: ${top}px`,
    `left: ${left}px`,
    `width: ${width}px`,
    `height: ${height}px`,
    `animation-duration: ${animationDuration}s`,
    `animation-delay: ${animationDelay}s`,
  ];

  return <MoonGlowHolder $styles={stylesArray} />;
}

export default MoonGlow;
