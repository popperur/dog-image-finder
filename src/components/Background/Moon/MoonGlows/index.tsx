import { MOON_GLOWS } from "components/Background/constants.ts";
import { MoonGlowContainer } from "components/Background/Moon/MoonGlows/styles.tsx";
import MoonGlow from "components/Background/Moon/MoonGlows/MoonGlow.tsx";

function MoonGlows() {
  const moonGlowsProps = MOON_GLOWS.map(
    ([top, left, width, height, animationDuration, animationDelay]) => ({
      top,
      left,
      width,
      height,
      animationDuration,
      animationDelay,
    }),
  );

  return (
    <MoonGlowContainer>
      {moonGlowsProps.map((moonGlowProps, index) => (
        <MoonGlow key={index} {...moonGlowProps} />
      ))}
    </MoonGlowContainer>
  );
}

export default MoonGlows;
