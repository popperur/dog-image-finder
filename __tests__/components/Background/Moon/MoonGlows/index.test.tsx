import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import { MOON_GLOWS } from "components/Background/constants.ts";
import MoonGlows from "components/Background/Moon/MoonGlows";

describe("MoonGlows component", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(<MoonGlows />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("shows all glows", () => {
    const { container } = render(<MoonGlows />);

    expect(container.firstChild?.childNodes).toHaveLength(MOON_GLOWS.length);
  });
});
