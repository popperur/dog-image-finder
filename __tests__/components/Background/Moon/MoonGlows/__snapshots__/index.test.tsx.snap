// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MoonGlows component > matches the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: absolute;
  background-color: #f1f1f1;
  border-radius: 50%;
  opacity: 0.1;
  transition: opacity 0.3s;
  animation: xvTUm infinite;
  top: 20px;
  left: 30px;
  width: 260px;
  height: 260px;
  animation-duration: 3s;
  animation-delay: 0s;
}

.c1 {
  position: absolute;
  background-color: #f1f1f1;
  border-radius: 50%;
  opacity: 0.1;
  transition: opacity 0.3s;
  animation: xvTUm infinite;
  top: -20px;
  left: -10px;
  width: 340px;
  height: 340px;
  animation-duration: 3s;
  animation-delay: 0.5s;
}

<div
    class=""
  >
    <div
      class="c0"
    />
    <div
      class="c1"
    />
  </div>
</DocumentFragment>
`;
